### DISCUSSION TITLE
Coming Soon: Caching Plugins for Laravel

### DISCUSSION THREAD

**Discussion by Sammyjo20** [saloonphp/saloon#51] [Announcements]
Just wanted to add an issue mentioning that I'm adding support for request caching in Saloon soon. It's going to use the popular Guzzle caching plugin [https://github.com/Kevinrob/guzzle-cache-middleware](https://github.com/Kevinrob/guzzle-cache-middleware) but I'm hoping to extend it slightly.

## Additions

- Adds **HasExplicitCaching** trait
- Adds **HasImplicitCaching** trait

### HasExplicitCaching
This caching plugin will allow you to define explicit caching rules, like how long a response should be cached for.

### HasImplicitCaching
This caching plugin will work automatically as it will use the caching headers (if provided) on the response.

Both plugins will allow you to purge the cache at any time and see if a SaloonResponse has been cached.

**Comment by Braunson** [comment]
This is an awesome plugin, I was about to write my own as I need this for a project. Looking forward to it.

**Comment by Sammyjo20** [comment]
Thanks @Braunson. My schedule for the next week is looking quite busy so not sure how much time I'll get to continue working on this but it's pretty much done, I just want to separate it into a separate plugin package and then it will be ready. In the meantime if you wanted to use the caching plugin, install https://github.com/Kevinrob/guzzle-cache-middleware and then add a Saloon handler which accepts this Plugin's middleware.

**Comment by Sammyjo20** [comment]
Apologies for the delay on this project. Work has become quite busy and I want to put all my time into that for the moment, but I have full hope to be back on Saloon in a few weeks time 🥳

**Comment by Braunson** [comment]
@Sammyjo20 No need, work pays the bills! 😊

**Comment by Sammyjo20** [comment]
The first version of the plugin is live! 🎉 There is a new section on the docs to check out, really happy with how it’s turned out so far. I will be working on a caching plugin that looks at the response’s cache headers but for now it’s manual
